"""trying fix issues with uuid generating

Revision ID: 7233ac53618d
Revises: 86c9314c3ebd
Create Date: 2024-11-24 21:03:45.020834

"""
from alembic import op
import sqlalchemy as sa
import uuid
from sqlalchemy.sql import table, column
from sqlalchemy import String

# revision identifiers, used by Alembic.
revision = '7233ac53618d'
down_revision = '86c9314c3ebd'
branch_labels = None
depends_on = None


users_table = table(
    'users',
    column('id', String),
    column('fs_uniquifier', String)
)

def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('users', schema=None) as batch_op:
        batch_op.alter_column('fs_uniquifier',
               existing_type=sa.VARCHAR(length=255),
               nullable=True)


    # Генерируем уникальные значения для fs_uniquifier для всех пользователей
    conn = op.get_bind()
    results = conn.execute(users_table.select())
    for row in results:
        conn.execute(
            users_table.update()
            .where(users_table.c.id == row.id)
            .values(fs_uniquifier=str(uuid.uuid4()))
        )
        
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('users', schema=None) as batch_op:
        batch_op.alter_column('fs_uniquifier',
               existing_type=sa.VARCHAR(length=255),
               nullable=False)

    # ### end Alembic commands ###
